src/cls/otp/cls_otp.cc

```diff
@@ -268,8 +268,8 @@ static int write_header(cls_method_context_t hctx, const otp_header& h)
 static int parse_seed(const string& seed, SeedType seed_type, bufferlist *seed_bin)
 {
   size_t slen = seed.length();
-  char secret[seed.length()];
-  char *psecret = secret;
+  std::vector<char> secret(seed.length());
+  char *psecret = secret.data();
```

Comment:
On line 295 there's a possibility of free() being called on this. Can the psecret pointer go away?

Reply:
Oh... So... *IF* need_free gets set, psecret is re-assigned (line 282) and then free()ed.